test/functional/feature_fee_estimation.py

```diff
@@ -125,61 +89,67 @@ def check_raw_estimates(node, fees_seen):
 
             if feerate + delta < min(fees_seen) or feerate - delta > max(fees_seen):
-                raise AssertionError(f"Estimated fee ({feerate}) out of range ({min(fees_seen)},{max(fees_seen)})")
+                raise AssertionError(
+                    f"Estimated fee ({feerate}) out of range ({min(fees_seen)},{max(fees_seen)})"
+                )
+
 
 def check_smart_estimates(node, fees_seen):
     """Call estimatesmartfee and verify that the estimates meet certain invariants."""
 
     delta = 1.0e-6  # account for rounding error
     last_feerate = float(max(fees_seen))
     all_smart_estimates = [node.estimatesmartfee(i) for i in range(1, 26)]
-    mempoolMinFee = node.getmempoolinfo()['mempoolminfee']
-    minRelaytxFee = node.getmempoolinfo()['minrelaytxfee']
+    mempoolMinFee = node.getmempoolinfo()["mempoolminfee"]
+    minRelaytxFee = node.getmempoolinfo()["minrelaytxfee"]
     for i, e in enumerate(all_smart_estimates):  # estimate is for i+1
         feerate = float(e["feerate"])
         assert_greater_than(feerate, 0)
         assert_greater_than_or_equal(feerate, float(mempoolMinFee))
         assert_greater_than_or_equal(feerate, float(minRelaytxFee))
 
         if feerate + delta < min(fees_seen) or feerate - delta > max(fees_seen):
-            raise AssertionError(f"Estimated fee ({feerate}) out of range ({min(fees_seen)},{max(fees_seen)})")
+            raise AssertionError(
+                f"Estimated fee ({feerate}) out of range ({min(fees_seen)},{max(fees_seen)})"
+            )
         if feerate - delta > last_feerate:
-            raise AssertionError(f"Estimated fee ({feerate}) larger than last fee ({last_feerate}) for lower number of confirms")
+            raise AssertionError(
+                f"Estimated fee ({feerate}) larger than last fee ({last_feerate}) for lower number of confirms"
+            )
         last_feerate = feerate
 
         if i == 0:
             assert_equal(e["blocks"], 2)
         else:
             assert_greater_than_or_equal(i + 1, e["blocks"])
 
+
 def check_estimates(node, fees_seen):
     check_raw_estimates(node, fees_seen)
     check_smart_estimates(node, fees_seen)
 
 
 def send_tx(node, utxo, feerate):
     """Broadcast a 1in-1out transaction with a specific input and feerate (sat/vb)."""
-    overhead, op, scriptsig, nseq, value, spk = 10, 36, 5, 4, 8, 24
-    tx_size = overhead + op + scriptsig + nseq + value + spk
-    fee = tx_size * feerate
-
     tx = CTransaction()
-    tx.vin = [CTxIn(COutPoint(int(utxo["txid"], 16), utxo["vout"]), SCRIPT_SIG[utxo["vout"]])]
-    tx.vout = [CTxOut(int(utxo["amount"] * COIN) - fee, P2SH_1)]
-    txid = node.sendrawtransaction(tx.serialize().hex())
+    tx.vin = [CTxIn(COutPoint(int(utxo["txid"], 16), utxo["vout"]), REDEEM_SCRIPT)]
+    tx.vout = [CTxOut(int(utxo["amount"] * COIN), P2SH)]
+
+    # vbytes == bytes as we are using legacy transactions
+    fee = tx.get_vsize() * feerate
+    tx.vout[0].nValue -= fee
 
```

Comment:
@darosior said "since we use legacy txs in this test it's indeed much cleaner" in https://github.com/bitcoin/bitcoin/pull/22539#discussion_r720917349. Does this mean for non-legacy txs, we can no longer use the same logic `     fee = tx.get_vsize() * feerate` to calculate fee?